### PR TITLE
Team fixes

### PR DESCRIPTION
**Pull request by skamoen**
This PR tweaks the Team functionality a bit to be more compatible with the front-end. A new Team View has been introduced, which displays additional information to Team members only. 

It also fixes a severe dataleak where Captains would get full member information on inviting a new member (very minor detail, really!).

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java [hunk 1]
```diff
@@ -156,9 +155,8 @@ public Team getTeamById(@PathVariable Long teamId) {
             "and @currentUserServiceImpl.hasAnyTicket(#username)")
```
Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java]: Does a user that is to be invited need to have a ticket? I know this isn't implemented in this PR but I haven't seen this before. I thought the idea was that you could form a team, and then buy tickets for your team.

File: src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java [hunk 2]
```diff
@@ -156,9 +155,8 @@ public Team getTeamById(@PathVariable Long teamId) {
...
     @RequestMapping(method = RequestMethod.POST, value = "/{teamId}/invites")
     public ResponseEntity<?> inviteTeamMember(@PathVariable Long teamId, @RequestBody String username) {
-        TeamInviteToken teamInviteToken = teamService.inviteMember(teamId, username);
-        return createResponseEntity(HttpStatus.OK, "User " + username + " successfully invited to Team " + teamId,
-                teamInviteToken);
+        teamService.inviteMember(teamId, username);
+        return createResponseEntity(HttpStatus.OK, "User " + username + " successfully invited to Team " + teamId);
```
Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java]: Why doesn't this part return the token anymore? Doesn't the frontend need the token to, for example, cancel an invite?

File: src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java [hunk 3]
```diff
@@ -167,17 +165,31 @@ public Team getTeamById(@PathVariable Long teamId) {
...
+    /**
+     * Decline or revoke an invite
+     *
+     * @param token Token of the Invite
+     *
+     * @return Statusmessage
+     */
     @PreAuthorize("@currentUserServiceImpl.canRevokeInvite(principal, #token)")
     @RequestMapping(method = RequestMethod.DELETE, value = "/invites")
     public ResponseEntity<?> declineTeamInvite(@RequestBody String token) {
```
Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java]: Revoke sounds as if the sender can also cancel the invite. I wouldn't name this method 'declineTeam...' but 'revokeTeam...'

File: src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java (hunk 1 above)
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java]: Yes, any Team related things require the user to have a Ticket. You can add users after you've sent them the ticket.

File: src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java (hunk 2 above)
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/controller/TeamRestController.java]: It returned the token itself, which contains a lot more information then you want to return. The Front-end requests the tokens when it needs them. That uses the `TokenResponse` object, which only contains the information required.

File: src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java [hunk 4]
```diff
@@ -89,8 +88,10 @@ public Team update(Long teamId, TeamDTO input) {
         }
 
         // If the Captain username is set, change the Captain
-        if (!Strings.isNullOrEmpty(input.getCaptainUsername())) {
-            User captain = userService.getUserByUsername(input.getCaptainUsername()).get();
+        String captainUsername = input.getCaptainUsername();
+        if (!Strings.isNullOrEmpty(captainUsername)) {
```
Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java]: I would also perform a check if the captain in input is that same as the current captain. I know this doesn't change anything in functionality but I think it's better code. Less execution.

File: src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java [hunk 5]
```diff
@@ -109,12 +110,17 @@ public TeamInviteToken inviteMember(Long teamId, String username) {
...
 
+        // Check if the member isn't already part of the team
+        if (team.getMembers().contains(user)) {
+            throw new IllegalArgumentException("User is already a member of this team");
+        }
+
         // Check if the member isn't already invited
-        Optional<TeamInviteToken> optionalInvite = teamInviteTokenRepository.findByUserUsername(username).stream().
-                filter(token -> token.getTeam().equals(team)).findFirst();
+        boolean userAlreadyInvited = teamInviteTokenRepository.findByUserUsername(username).stream()
```
Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java]: I would extract this logic (as well as some other logic in this method) in seperate methods. This entire method is 30 lines long. In my opinion way too long. This does not only provide better code readability but also code reusability.

**Comment by bvanderdrift** [comment]
Back to @skamoen

**Comment by skamoen** [comment]
Updated. Back to @bvanderdrift

**Comment by bvanderdrift** [comment]
Looking good!

**Pull request merged**